contrib/storage-splunk/src/main/java/org/apache/drill/exec/store/splunk/SplunkConnection.java

```diff
@@ -41,11 +41,14 @@
   private final String hostname;
   private final int port;
   private Service service;
+  private Integer reconnectRetries;
+  private int counter;
```

Comment:
We could use a single variable here, for example, `connectionAttempts`, decrease its value every time we are trying to connect and check in the catch block whether it is larger than 0.